Build fail

Following the `README.md` with  the main `LLVM-MCA-Daemon` branch and upstream `llvm-project`following steps:
```
1. git clone https://github.com/llvm/llvm-project.git

2. cd llvm-project

4. git checkout 56ffbd97fda16008d02180a460211829354f1094 (according to latest COMMIT file its supported)

5.git am /path/to/LLVM-MCA-Daemon/patches/*.patch  # Important: add custom LLVM patches

6. mkdir install

7. mkdir build && cd build

8. cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug \
               -DCMAKE_INSTALL_PREFIX=$(realpath ../install) \
               -DBUILD_SHARED_LIBS=ON \
               -DLLVM_TARGETS_TO_BUILD="X86;ARM;PowerPC" \
               -DLLVM_USE_LINKER=gold \
               -DLLVM_DEFAULT_TARGET_TRIPLE=x86_64-pc-linux-gnu \
               ../llvm

9. ninja llvm-mca llvm-mc LLVMDebugInfoDWARF
10. ninja install

11. cd ../../LLVM-MCA-Daemon/

12. mkdir .build && cd .build

13. cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug \
               -DLLVM_DIR=$(realpath ../../llvm-project/install/lib/cmake/llvm) \
               -DQEMU_INCLUDE_DIR=$(realpath ../../qemu/include) \
               -DLLVM_MCAD_ENABLE_PLUGINS=qemu \
               ../

14. ninja all
```

The last ninja command leads to the following build error 

```
FAILED: CMakeFiles/llvm-mcad.dir/CustomHWUnits/MCADLSUnit.cpp.o 
/usr/bin/c++ -DGTEST_HAS_RTTI=0 -I/specific/disk1/hodbadihi/llvm-project/install/include -I/specific/disk1/hodbadihi/LLVM-MCA-Daemon -fno-rtti -g -std=gnu++17   -D_GNU_SOURCE -D_DEBUG -D_GLIBCXX_ASSERTIONS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -fno-exceptions -funwind-tables -fno-rtti -MD -MT CMakeFiles/llvm-mcad.dir/CustomHWUnits/MCADLSUnit.cpp.o -MF CMakeFiles/llvm-mcad.dir/CustomHWUnits/MCADLSUnit.cpp.o.d -o CMakeFiles/llvm-mcad.dir/CustomHWUnits/MCADLSUnit.cpp.o -c /specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp
In file included from /specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:7:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:73:61: error: expected class-name before ‘{’ token
   73 |   class CustomMemoryGroup : public mca::LSUnit::MemoryGroup {
      |                                                             ^
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:126:8: error: ‘bool llvm::mcad::MCADLSUnit::isReady(const llvm::mca::InstRef&) const’ marked ‘override’, but does not override
  126 |   bool isReady(const mca::InstRef &IR) const override {
      |        ^~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:132:8: error: ‘bool llvm::mcad::MCADLSUnit::isPending(const llvm::mca::InstRef&) const’ marked ‘override’, but does not override
  132 |   bool isPending(const mca::InstRef &IR) const override {
      |        ^~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:138:8: error: ‘bool llvm::mcad::MCADLSUnit::isWaiting(const llvm::mca::InstRef&) const’ marked ‘override’, but does not override
  138 |   bool isWaiting(const mca::InstRef &IR) const override {
      |        ^~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:144:8: error: ‘bool llvm::mcad::MCADLSUnit::hasDependentUsers(const llvm::mca::InstRef&) const’ marked ‘override’, but does not override
  144 |   bool hasDependentUsers(const mca::InstRef &IR) const override {
      |        ^~~~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:151:3: error: ‘const llvm::mca::CriticalDependency llvm::mcad::MCADLSUnit::getCriticalPredecessor(unsigned int)’ marked ‘override’, but does not override
  151 |   getCriticalPredecessor(unsigned GroupId) override {
      |   ^~~~~~~~~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:170:16: error: ‘virtual void llvm::mcad::MCADLSUnit::dump() const’ marked ‘override’, but does not override
  170 |   virtual void dump() const override;
      |                ^~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h: In member function ‘bool llvm::mcad::MCADLSUnit::isReady(const llvm::mca::InstRef&) const’:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:129:18: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘isReady’
  129 |     return Group.isReady() && !isInstOperandWaitingForCache(IR);
      |                  ^~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h: In member function ‘bool llvm::mcad::MCADLSUnit::isPending(const llvm::mca::InstRef&) const’:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:135:18: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘isPending’
  135 |     return Group.isPending();
      |                  ^~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h: In member function ‘bool llvm::mcad::MCADLSUnit::isWaiting(const llvm::mca::InstRef&) const’:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:141:18: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘isWaiting’
  141 |     return Group.isWaiting() || isInstOperandWaitingForCache(IR);
      |                  ^~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h: In member function ‘bool llvm::mcad::MCADLSUnit::hasDependentUsers(const llvm::mca::InstRef&) const’:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:147:19: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘isExecuted’
  147 |     return !Group.isExecuted() && Group.getNumSuccessors();
      |                   ^~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:147:41: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘getNumSuccessors’
  147 |     return !Group.isExecuted() && Group.getNumSuccessors();
      |                                         ^~~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h: In member function ‘const llvm::mca::CriticalDependency llvm::mcad::MCADLSUnit::getCriticalPredecessor(unsigned int)’:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:153:18: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘getCriticalPredecessor’
  153 |     return Group.getCriticalPredecessor();
      |                  ^~~~~~~~~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h: In member function ‘virtual void llvm::mcad::MCADLSUnit::onInstructionIssued(const llvm::mca::InstRef&)’:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.h:160:28: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘onInstructionIssued’
  160 |     CustomGroups[GroupID]->onInstructionIssued(IR);
      |                            ^~~~~~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp: In member function ‘virtual unsigned int llvm::mcad::MCADLSUnit::dispatch(const llvm::mca::InstRef&)’:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:151:14: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘addInstruction’
  151 |     NewGroup.addInstruction();
      |              ^~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:164:12: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘addSuccessor’
  164 |       IDom.addSuccessor(&NewGroup, !noAlias(ImmediateLoadDominator, MaybeMDA));
      |            ^~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:174:18: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘addSuccessor’
  174 |       StoreGroup.addSuccessor(&NewGroup, true);
      |                  ^~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:183:18: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘addSuccessor’
  183 |       StoreGroup.addSuccessor(&NewGroup,
      |                  ^~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:221:46: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘isExecuting’
  221 |       getCustomGroup(ImmediateLoadDominator).isExecuting();
      |                                              ^~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:226:14: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘addInstruction’
  226 |     NewGroup.addInstruction();
      |              ^~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:238:18: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘addSuccessor’
  238 |       StoreGroup.addSuccessor(&NewGroup, true);
      |                  ^~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:247:19: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘addSuccessor’
  247 |         LoadGroup.addSuccessor(&NewGroup, true);
      |                   ^~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:257:19: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘addSuccessor’
  257 |         LoadGroup.addSuccessor(&NewGroup, true);
      |                   ^~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:269:9: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘addInstruction’
  269 |   Group.addInstruction();
      |         ^~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp: In member function ‘virtual void llvm::mcad::MCADLSUnit::onInstructionExecuted(const llvm::mca::InstRef&)’:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:321:15: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘onInstructionExecuted’
  321 |   It->second->onInstructionExecuted(IR);
      |               ^~~~~~~~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:322:19: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘isExecuted’
  322 |   if (It->second->isExecuted())
      |                   ^~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp: In member function ‘virtual void llvm::mcad::MCADLSUnit::cycleEvent()’:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:339:15: error: ‘class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘cycleEvent’
  339 |     G.second->cycleEvent();
      |               ^~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp: In member function ‘virtual void llvm::mcad::MCADLSUnit::dump() const’:
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:355:38: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘getNumPredecessors’
  355 |            << "[ #Preds = " << Group.getNumPredecessors()
      |                                      ^~~~~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:356:40: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘getNumExecutingPredecessors’
  356 |            << ", #GIssued = " << Group.getNumExecutingPredecessors()
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:357:42: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘getNumExecutedPredecessors’
  357 |            << ", #GExecuted = " << Group.getNumExecutedPredecessors()
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:358:37: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘getNumInstructions’
  358 |            << ", #Inst = " << Group.getNumInstructions()
      |                                     ^~~~~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:359:40: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘getNumExecuting’
  359 |            << ", #IIssued = " << Group.getNumExecuting()
      |                                        ^~~~~~~~~~~~~~~
/specific/disk1/hodbadihi/LLVM-MCA-Daemon/CustomHWUnits/MCADLSUnit.cpp:360:42: error: ‘const class llvm::mcad::MCADLSUnit::CustomMemoryGroup’ has no member named ‘getNumExecuted’
  360 |            << ", #IExecuted = " << Group.getNumExecuted() << '\n';
      |                                          ^~~~~~~~~~~~~~
[73/2921] Building CXX object CMakeFiles/llvm-mcad.dir/Brokers/AsmUtils/CodeRegionGenerator.cpp.o
```

I also attempted a combination of the `broker-improvements` and `dev-incremental-mca` branches, as suggested in another issue. However, this produced a different error. I suspect a version mismatch is the cause and would welcome any guidance on resolving it

--- a/Brokers/AsmUtils/CodeRegionGenerator.cpp
+++ b/Brokers/AsmUtils/CodeRegionGenerator.cpp
@@ -63,7 +63,6 @@ public:
   void emitZerofill(MCSection *Section, MCSymbol *Symbol = nullptr,
                     uint64_t Size = 0, Align ByteAlignment = Align(1),
                     SMLoc Loc = SMLoc()) override {}
-  void emitGPRel32Value(const MCExpr *Value) override {}
   void BeginCOFFSymbolDef(const MCSymbol *Symbol) {}
   void EmitCOFFSymbolStorageClass(int StorageClass) {}
   void EmitCOFFSymbolType(int Type) {}

